### ISSUE TITLE
View options inconsistency

### ISSUE THREAD

**Issue by paulfalgout** [marionettejs/backbone.marionette#2242]
This issue is similar to https://github.com/marionettejs/backbone.marionette/issues/2229

When adding options to a `Marionette.View` you can either add them to the `View` definition:
```
var MyView = Marionette.View.extend({
  options: {}  // can also be a function returning a hash
});
```
or you can pass the options when instantiating:
```
var options = {};  //can also be a function returning a hash
var myView = new Marionette.View(options);
```
In both cases they are added to `myView.options` here: 
https://github.com/marionettejs/backbone.marionette/blob/master/lib/backbone.marionette.js#L1632

**Backbone Views**
Backbone Views used to attach options in a very similar way.  https://github.com/jashkenas/backbone/blob/1.0.0/backbone.js#L1087
But this functionality was removed in `v1.1.0`

**The issue**
Marionette does attach `options` to the `View` as Backbone used to, however it does not pass the `options` hash to the `Backbone.View` `constructor`.  So if you add options to the `View`'s definition as in the first example, or if you return `options` as a function when passing them during instantiation, you cannot expect the `Backbone.View` to receive these `options`.

**Examples**
```
// As Expected

var options = {
  className: '.some-class'
};  

var myView = new Marionette.View(options);

console.log(myView.options.className === '.some-class'); // -> logs true
console.log(myView.className === '.some-class'); // -> logs true
```

```
// Unexpected

var MyView = Marionette.View.extend({
  options: {
    className: '.some-class'
  }  // can also be a function returning this hash
});

var myView = new MyView();

console.log(myView.options.className === '.some-class'); // -> logs true
console.log(myView.className === '.some-class'); // -> logs false
```

As in https://github.com/marionettejs/backbone.marionette/issues/2229
```
// Unexpected

var options = function(){
  return {
    className: '.some-class'
  };
};  

var myView = new Marionette.View(options);

console.log(myView.options.className === '.some-class'); // -> logs true
console.log(myView.className === '.some-class'); // -> logs false
```

**Possible solutions**

1. We could remove the ability for `options` to be a function and/or for `options` to be attached to the `View` definition.  I think I prefer both of these in the long term.  So essentially just `this.options = options || {};`

2. We could `Backbone.View.call(this, options);`  (Or some version of modifying the `options` on `arguments`) instead of `Backbone.View.apply(this, arguments);`  We would have to do this for each view type.  This way the result of all possible `options` get sent to `Backbone.View`.  I suppose this is potentially breaking of something somewhere.. however you'll find that `LayoutView` uses `call` and `ItemView`, `CollectionView`, and `CompositeView` use `apply` so either way it looks like we may have a consistency issue.  

3. We could document the inconsistencies.

A related issue is that `CollectionView` looks for `sort` on `options`, but does not evaluate `options` as a function.  It does however look at `sort` on the `View` definition, but does not look at the `View` definition's option hash for a possible `sort` field.  So similarly to the Backbone viewOptions, you must pass `options` as a hash on the instantiation of a `CollectionView` in order for it to work if using `options`.
https://github.com/marionettejs/backbone.marionette/blob/master/lib/backbone.marionette.js#L2063

**Comment by jasonLaster** [comment]
Great find, better write up. This would have been hard to follow if you weren't clear in the overview and examples. 

I'd like to see some failing tests and the necessary patch so that everything works as expected. As for doing away with options you could make the argument that we should, but if this is a fixable problem I'd like to separate the discussion. Doing away with options is a design decision this is a compliance discussion.

**Comment by paulfalgout** [comment]
Just for clarification, I don't want to do away with options.  I actually like the functionality.  But I do think passing options as a function or defining options on the View's definition are only semi-implemented, probably long forgotten features that could be, likely should be removed.

That said I'm all up for fixing it in the near term with discussions for removal in the future, but is there any concern about moving from `Marionette.View.apply` to `Marionette.View.call` in the various constructors?  Seems like it'd be fine...

**Comment by jasonLaster** [comment]
should be okay
